AzureIoTHub.agent.lib.nut

```diff
@@ -1177,11 +1232,45 @@ class AzureIoTHub {
             _topics.dMethodResp <- "$iothub/methods/res/";
         }
 
+        function _updatePasswd() {
+            local sasExpTime = time() + _options.tokenTTL;
+            local sas = AzureIoTHub.SharedAccessSignature(_resourceUri, null, _connStrParsed.SharedAccessKey, sasExpTime).toString();
+            _tokenExpiresAt = sasExpTime;
+            _mqttOptions.password = sas;
+        }
+
         function _onConnected(err) {
+            if (_isRefreshingToken) {
+                if (err == 0) {
+                    _log("Reconnected with new token!");
+                    local resubscribed = function(err) {
+                        _isRefreshingToken = false;
+                        if (err == 0) {
```

Comment:
It must be the internal `err`, not the external one, correct? Might be working well, but looks very confusing. I would probably give them different names.